ext/standard/var_unserializer.re

```diff
@@ -1201,6 +1201,114 @@ object ":" uiv ":" ["]	{
 	return object_common(UNSERIALIZE_PASSTHRU, elements, has_unserialize);
 }
 
+"E:" uiv ":" ["]	{
+	size_t len, maxlen;
+	char *str;
+	zend_string *full_name, *enum_name, *case_name;
+	zend_class_entry *ce;
+
+	if (!var_hash) return 0;
+
+	len = parse_uiv(start + 2);
+	maxlen = max - YYCURSOR;
+	if (maxlen < len || len == 0) {
+		*p = start + 2;
+		return 0;
+	}
+
+	str = (char*)YYCURSOR;
+
+	YYCURSOR += len;
+
+	if (*(YYCURSOR) != '"') {
+		*p = YYCURSOR;
+		return 0;
+	}
+	if (*(YYCURSOR+1) != ';') {
+		*p = YYCURSOR+1;
+		return 0;
+	}
+
+	full_name = zend_string_init(str, len, 0);
+	int32_t colon_pos = -1;
+	for (uint32_t i = 0; i < ZSTR_LEN(full_name); ++i) {
+		if (ZSTR_VAL(full_name)[i] == ':') {
+			colon_pos = i;
+			break;
+		}
+	}
+	if (colon_pos == -1) {
+		zend_string_release_ex(full_name, 0);
+		php_error_docref(NULL, E_WARNING, "Invalid enum name '%s' (missing colon)", ZSTR_VAL(full_name));
```

Comment:
ZSTR_VAL is just a macro to return a C string (`char*`), so it's still easy to avoid allocating and freeing full_name:

```suggestion
		php_error_docref(NULL, E_WARNING, "Invalid enum name '%s' (missing colon)", str);
```

Reply:
Never mind, `str` wouldn't be null terminated because it's part of a larger buffer, so `%.*s...", (int)len, str)` wouldn't be fully the same either (error messages typically wouldn't include null bytes of binary data).

Could still allocate it only for the error message

Reply:
Yes but `str` is not `NULL` terminated. Maybe we can do something with `%.*s`.

Reply:
```c
    printf("val='%.*s'\n", 10, "mixed\0after\0"); // prints val='mixed'
```
Seems like it works, and %.*s will stop either at the `precision` or the specified `len`

Reply:
Sorry, didn't reload the page so didn't see your comment. `%.*s` seems to work :+1: